internal/transformer/projector.go

```diff
@@ -13,13 +17,17 @@ import (
 type Projector struct {
 	projector pkg.Projector
 	log       logger.Log
+	collector instrument.Collector
+	collect   bool
```

Comment:
Actually when no collector is defined i am using a Noop Collector which does nothing.
To avoid `if collector != nil` in many places (because collector code is in many places)
But you are right i will try to find a way to remove this boolean...

just set the collector to noop in this case actually
👍